pkg/trace/api/otlp.go

```diff
@@ -201,15 +202,15 @@ func fastHeaderGet(h http.Header, canonicalKey string) string {
 
 // processRequest processes the incoming request in. It marks it as received by the given protocol
 // using the given headers.
-func (o *OTLPReceiver) processRequest(protocol string, header http.Header, in ptraceotlp.Request) {
+func (o *OTLPReceiver) processRequest(ctx context.Context, protocol string, header http.Header, in ptraceotlp.Request) {
 	for i := 0; i < in.Traces().ResourceSpans().Len(); i++ {
 		rspans := in.Traces().ResourceSpans().At(i)
-		o.ReceiveResourceSpans(rspans, header, protocol)
+		o.ReceiveResourceSpans(ctx, rspans, header, protocol)
 	}
 }
 
 // ReceiveResourceSpans processes the given rspans and returns the source that it identified from processing them.
-func (o *OTLPReceiver) ReceiveResourceSpans(rspans ptrace.ResourceSpans, header http.Header, protocol string) source.Source {
+func (o *OTLPReceiver) ReceiveResourceSpans(ctx context.Context, rspans ptrace.ResourceSpans, header http.Header, protocol string) source.Source {
```

Comment:
Since this is specifically the only API that the [Datadog Exporter uses](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.55.0/exporter/datadogexporter/traces_exporter.go#L127) in the OpenTelemetry Collector, I wonder if we should report this as a breaking change. But probably not. Leaving this comment here as more of note to myself to change this when we update that.